virtual/registry/kv_registry.go

```diff
@@ -431,15 +434,28 @@ func (k *kvRegistry) Heartbeat(
 
 		var state serverState
 		if !ok {
+			serverVersion = 0
 			state = serverState{
-				ServerID: serverID,
+				ServerID:      serverID,
+				ServerVersion: serverVersion,
 			}
+			state.LastHeartbeatedAt = time.Now()
```

Comment:
This is fine for now, but can you file an issue to replace all usage of `time.Now()` and `time.Since()` with `tr.getVersionStamp()` ? We shouldn't be relying on each machine's clock to be accurate, we should be using the registry's clock (the versionstamp)

Reply:
Alternative you can refactor it in your p.r. `time.Now()` -> `tr.getVersionStamp()`

`time.Since()` -> 

```golang
func versionSince(tr, prev) time.Duration {
    var (
        curr := tr.getVersionStamp()
        since := curr-compareTo
    )
    if since < 0 {
         panic("compareTo: %d, curr: %d, versionstamp did not increase monotonically", prev, curr)
    }
    return time.Duration(since*time.Microsecond)
}
```

This should work because all implementations of `getVersionStamp` are designed to increase at a rate of ~1 per microsecond (or 1 million per second like FDB's does).

Reply:
My code may not compile / be 100% right but it should be pretty close